modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1568,6 +1576,33 @@ ABSDIFFC_SIMD(float)
 
 #undef ABSDIFFC_SIMD
 
+//-------------------------
+//
+// Fluid kernels: Split3
+//
+//-------------------------
+
+#define SPLIT3_SIMD(SRC, DST)                                       \
+int split3_simd(const SRC in[], DST out1[], DST out2[],             \
+                DST out3[], const int width)                        \
+{                                                                   \
+    constexpr int nlanes = v_uint8::nlanes;                         \
+    int x = 0;                                                      \
+    for (; x <= width - nlanes; x += nlanes)                        \
+    {                                                               \
+        v_uint8 a, b, c;                                            \
```

Comment:
It would be better to move creation and initialization of these variables from the loop.